src/frontend/src/lib/components/dapps/DappsExplorerSignedIn.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import DappCard from '$lib/components/dapps/DappCard.svelte';
+	import DappModal from '$lib/components/dapps/DappModal.svelte';
+	import DappPromoBanner from '$lib/components/dapps/DappPromoBanner.svelte';
+	import Button from '$lib/components/ui/Button.svelte';
+	import { modalDAppDetails } from '$lib/derived/modal.derived';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import {
+		type DappDescription,
+		dAppDescriptions,
+		type FeaturedDappDescription
+	} from '$lib/types/dappDescription';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let selectedTag: string | null = null;
+	const featuredDapp: FeaturedDappDescription | undefined = dAppDescriptions.find(
+		(dApp) => dApp.featured && nonNullish(dApp.screenshots) && dApp.screenshots.length > 0
+	) as FeaturedDappDescription | undefined;
+
+	const uniqueTags = new Set(
+		dAppDescriptions.flatMap((dapp) => dapp.tags).sort((tagA, tagB) => tagA.localeCompare(tagB))
+	);
+
+	$: filteredDapps = nonNullish(selectedTag)
+		? dAppDescriptions.filter((dApp) => dApp.tags.includes(selectedTag!))
```

Comment:
Please avoid the use of `!`.

cc @AntonioVentilii-DFINITY can you setup an eslint rule?

Reply:
Ah I see, it lints, but my idea does not like it. Presumably because of the nonNullish wrapper.. 
<img width="860" alt="image" src="https://github.com/user-attachments/assets/f815d715-0795-4d9d-b4c7-f849defe24b9">

Reply:
just wondering.. I had to apply a cast to the selectedTag because the npm run check did not succeed. Is this the normal case when using nonNullish? Using the normal comparison, the compiler would be happy.
<img width="1087" alt="image" src="https://github.com/user-attachments/assets/35be700a-bf21-4ae7-bd29-7a356ed4f454">

Reply:
It seems unrelated to `nonNullish`. If you `selectedTag !== undefined` the linter complains the same way. Tried last version of TypeScript, same same. I guess it cannot defers it.

```
Error: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'. (ts)
		selectedTag !== undefined
			? dAppDescriptions.filter((dApp) => dApp.tags.includes(selectedTag))
			: dAppDescriptions;


====================================
svelte-check found 1 error and 0 warnings in 1 file
```

Let's add an additional test instead of a cast:

```
$: filteredDapps = nonNullish(selectedTag)
		? dAppDescriptions.filter(({ tags }) => nonNullish(selectedTag) && tags.includes(selectedTag))
		: dAppDescriptions;
```

Or a bit less performant but code wise more concise:

```
$: filteredDapps = dAppDescriptions.filter(
		({ tags }) => isNullish(selectedTag) || tags.includes(selectedTag)
	);
```

Reply:
gotcha, you're right, I forgot to remove the bang (!) 😄